### ISSUE TITLE
Dynmap not being excluded from backups

### ISSUE THREAD

**Issue by DR4GONSTEAR** [hexparrot/mineos-node#541]
Original "[Dynmap not being excluded from backups](https://github.com/hexparrot/mineos-node/issues/366)"
Code from 06/08/2020
`var args = ['--exclude', path.join(self.env.cwd, 'dynmap'), '{0}/'.format(self.env.cwd), self.env.bwd];`

Code from mineos.js 13/03/2024
`var args = ['--exclude', path.join(self.env.cwd, 'dynmap'), '{0}/'.format(self.env.cwd), self.env.bwd];`
I was hoping this would have been changed in the last 3 years but the code hasn't changed at all, and still backs up dynmap maps in /var/games/minecraft/[world]/plugins/dynmap/web/tiles

I tried messing around with this code and instead implementing:
`var args = ['--exclude', path.join(self.env.cwd, 'plugins/dynmap/web/tiles'), '--exclude', '*/dynmap', '{0}/'.format(self.env.cwd), self.env.bwd];`
But it failed to run any backups.

I was thinking this should work:
`var args = ['--exclude', path.join(self.env.cwd, 'plugins/dynmap/web/tiles'), '{0}/'.format(self.env.cwd), self.env.bwd];`
Shouldn't this backup config files, but not the actual map pictures that inflate even a small 2 person server to +10's of GB? But I'm not a coder, and I am unsure if there are more changes to the code I'd need to make. And I can't keep pulling my server down for testing while people are trying to play.

Long story short, I tried to change it myself, but I need help understanding if this is all that needs changing, or if it also needs to be updated somewhere else. Thoughts?

**Comment by DR4GONSTEAR** [comment]
Change this line:
`    var args = ['--exclude', path.join(self.env.cwd, 'dynmap'), '{0}/'.format(self.env.cwd), self.env.bwd];`

To these lines:
`    var excludedFolder = path.join(self.env.cwd, 'servers', '*', 'plugins', 'dynmap', 'web', 'tiles');`
`    var args = ['--exclude', excludedFolder, '{0}/'.format(self.env.cwd), self.env.bwd];`

This will stop the backup from including /plugins/dynmap/web/tiles/ which will enable backing up config files, but not the maps.

However I cannot achieve the same for archive and archive_with_commit
I have tried to change the two instances of this line:
`    var args = ['czf', path.join(self.env.awd, filename), '.'];`

To this line:
`    var args = ['czf', path.join(self.env.awd, filename), '--exclude', 'plugins/dynmap/web/tiles/', '.'];`

But it appears to do nothing, it still archives /plugins/dynmap/web/tiles/. Should I be using the same syntax as backup? I can't find any info either way but neither works.
`    var args = ['czf', path.join(self.env.awd, filename), '--exclude', 'servers', '*', 'plugins', 'dynmap', 'web', 'tiles', '.'];`

Am I on the right track, or do I need to add or change something else?

**Comment by DR4GONSTEAR** [comment]
I've since realised that archive_with_commit does nothing.

Changing all instances of the line:
`    var args = ['czf', path.join(self.env.awd, filename), '.'];`
To:
`    var args = ['czf', path.join(self.env.awd, filename), '--exclude=./plugins/dynmap/web/tiles', '.'];`
would be preferred in that case.

**Comment by DR4GONSTEAR** [comment]
It didn't even occur to me, but this is true for Paper servers, and its location of plugins/dynmap/web/tiles. If the plugin folder is ever located elsewhere, this would fail to omit the map files in a backup.
